### PR TITLE
updated the delete endpoint

### LINKED ISSUE #3111: [Dev] Update APD delete endpoint to incorporate Mongo
Update the delete endpoint to use Mongo instead of Postgress.

### This task is done when…
 - [ ] the delete endpoint has been updated to delete APDs from Mongo
 - [ ] the tests are updated

### PR DESCRIPTION
**Pull request by thetif**
resolves #3111 

**Description-**
Updated the delete endpoint to use mongo instead of postgres

**This pull request changes...**

- updates the delete endpoint to use mongo instead of postgres
- updated the tests to work with the new set up
- adds delete back into openapi

**Steps to manually verify this change...**

1. download the branch
2. set the following envs
```
export MONGO_URL="mongodb://mongo:cms@localhost:27017/eapd"
export MONGO_DATABASE="eapd"
```
3. cd to api folder
4. npm test
5. ./endpoint-tests/endpoint.sh
6. npm run lint
7. npm run audit
8. All of those should pass

### This pull request is ready to review when...

- [X] Automated tests are updated (and all tests are passing)
- [X] The change has been documented
- [X] Associated OpenAPI documentation has been updated
- [X] Changelog is updated as appropriate

### This pull request can be merged when…

- [ ] Code has been reviewed by someone other than the original author
- [ ] QA has verified the accessibility and functionality related to the change
- [ ] Design has approved the experience
- [ ] Product has approved the experience

### CODE REVIEW

File: api/db/apds.js [hunk 1]
```diff
@@ -9,13 +9,10 @@ const createAPD = async (apd, { APD = mongoose.model('APD') } = {}) => {
...
 
-const deleteAPD = async (id, { APD = mongoose.model('APD') } = {}) => {
-  const result = await APD.deleteOne({ _id: id });
-  return result;
-};
-
-const deleteAPDByID = async (id, { db = knex } = {}) => {
-  await db('apds').where('id', id).update({ status: 'archived' });
+const deleteAPDByID = async (id, { APD = mongoose.model('APD') } = {}) => {
+  const apd = await APD.findById(id).exec();
```
Comment (tbolt) [inline api/db/apds.js:13]: Why not use a `.findByIdAndUpdate()` here?

**Review by tbolt — APPROVED** [review]

File: api/db/apds.js (hunk 1 above)
Comment (thetif) [reply api/db/apds.js:13]: good call, I'll updated it

Comment (thetif) [reply api/db/apds.js]: I updated it to be more streamlined, but findByIdAndUpdate is deprecated so I went with updateOne

**Review by knollfear — APPROVED** [review]
LGTM.  Local tests all passed

**Pull request merged**